src/measure_theory/integration.lean

```diff
@@ -1831,6 +1838,37 @@ measure.of_measurable (λs hs, ∫⁻ a in s, f a ∂μ) (by simp) (λ s hs hd,
   μ.with_density f s = ∫⁻ a in s, f a ∂μ :=
 measure.of_measurable_apply s hs
 
+lemma with_density_add {f g : α → ℝ≥0∞} (hf : measurable f) (hg : measurable g) :
+  μ.with_density (f + g) = μ.with_density f + μ.with_density g :=
+begin
+  refine measure_theory.measure.ext (λ s hs, _),
+  rw [with_density_apply _ hs, measure.coe_add, pi.add_apply,
+      with_density_apply _ hs, with_density_apply _ hs, ← lintegral_add hf hg],
+  refl,
+end
+
+lemma with_density_smul (r : ℝ≥0∞) {f : α → ℝ≥0∞} (hf : measurable f) :
```

Comment:
I suspect the measurability of `f` is not required for this one.

Reply:
No, sorry for r = top it is probably necessary.

Reply:
I've added a version without measurability for `r \neq \infty`

Reply:
> No, sorry for r = top it is probably necessary.

Right, for r = top, I think the rhs = zero while the lhs = top for nonmeasurable `f`